command/healthcheck/pki.go

```diff
@@ -227,3 +227,49 @@ func pkiFetchLeaf(e *Executor, serial string, versionError func()) (bool, *PathF
 
 	return false, leafRet, leafRet.ParsedCache["certificate"].(*x509.Certificate), nil
 }
+
+func pkiFetchRoles(e *Executor, versionError func()) (bool, *PathFetch, []string, error) {
```

Comment:
nit: might be better to call out these are just role names and not the actual roles coming back?

```suggestion
func pkiFetchRoleNames(e *Executor, versionError func()) (bool, *PathFetch, []string, error) {
```

Reply:
Should I rename all the other helpers? We have `pkiFetchIssuers` -> `pkiFetchIssuer`, `pkiFetchLeaves` -> `pkiFetchLeaf`, and here `pkiFetchRoles` -> `pkiFetchRole`. Thoughts?

Reply:
Your call, it would make things clearer but not a huge gain either way.

Reply:
Yeah, the types already differ (the latter always take an issuer/serial/role reference) versus these take just two params, but I could push that in a later PR.

Reply:
Pushed in #18017